README.md

```diff
@@ -48,16 +55,106 @@ jobs:
+When you merge this PR ⚡️:
+
+- 🤖 It will request an Npm OTP from Optic. (If you close the PR, nothing will happen)
+- 🤖 Upon successful retrieval of the OTP, it will publish the package to Npm.
+- 🤖 Create a Github release with change logs (You can customize release notes using [release.yml](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#example-configuration))
+
+
+#### How to add a build step to your workflow
+
+When your project need a build step, you can run it before the release.
+
+Mainly, there are two build types:
+
+- to be committed: the build output should be committed to the repository and appear in the PR opened by this action
+- to be released: the build output should not be committed but its output should be published to Npm
+
+To do it you need to add some extra steps to your workflow.
+
+##### To be committed
+
+In this example, the `npm run build` command will create some artifacts that will be committed to the repository.
+Of course, you can use any build tool you need.
+_These artifacts should not be listed in the `.gitignore` file._
+
+The action will add the changes in the new release PR for you!
+
+```yml
+name: build-and-release
+
+on:
+  workflow_dispatch:
+    inputs:
+      semver:
+        description: "The semver to use"
+        required: true
+        default: "patch"
+  pull_request:
+    types: [closed]
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: install your runtime
+        uses: actions/setup-node@v2.5.0
+        with:
+          node-version: 16
+
+      - name: build the project when the PR is being opened
+        if: ${{ github.event_name == 'workflow_dispatch' }}
+        run: |
+          npm install
+          npm run build
+
+      - uses: nearform/optic-release-automation-action@v2
+        with:
+          github-token: ${{secrets.github_token}}
+          npm-token: ${{secrets.NPM_TOKEN}}
+          optic-token: ${{secrets.OPTIC_TOKEN}}
+          semver: ${{ github.event.inputs.semver }}
+```
+
+Now your PR includes the build artifacts, so when you will merge the PR, the action will publish the artifacts to Npm as expected.
+
+_Another approach is to commit manually the artifacts you want to release into the release branch._
+
+
+##### To be released
+
+You may want to release the artifacts without committing them to the repository such as a `dist/` folder.
+In this case, your workflow will look similar to the previous one, but the build step needs to run when the PR is merged:
+
+```yml
+      # [...] same workflow as before ...
+
+      - name: build the project when the PR is merged
+        if: ${{ github.event_name == 'pull_request' }}
+        run: |
+          npm install
+          npm run build
```

Comment:
ok so now I have a doubt about the approach. let's say that we want to embed the version number *after* the bump inside the application, we might do that by reading the package.json file *after* the bump during in the build script. What that means is that the build script would need to be invoked *after* the bump is done. Because the bump is done I'm now wondering if this approach of externalizing npm i / npm run build outside the action simply wouldn't work, and instead we have to use the approach used by cypress (as described in the issue)

Reply:
You are right, this case can be covered - I need to think a bit about how we could manage it for the user.

I may check deeper the cypress use case if there are some downside